File: backend/Booksy/src/main/java/hr/team16/booksy/model/Room.java
package hr.team16.booksy.model;

import jakarta.persistence.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@Entity
@Table(name = "room")
public class Room {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "property_id", nullable = false)
    private Property property;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(nullable = false)
    private Integer capacity = 2;

    @Column(name = "price_per_night", nullable = false, precision = 10, scale = 2)
    private BigDecimal pricePerNight;

    @Column(name = "available_from", nullable = false)
    private LocalDate availableFrom;

    @Column(name = "available_to", nullable = false)
    private LocalDate availableTo;
}

File: mobile/app/src/main/java/hr/team16/booksy/model/Room.kt
package hr.team16.booksy.model

import java.math.BigDecimal

data class Room(
    val id: Long,
    val name: String,
    val capacity: Int,
    val pricePerNight: BigDecimal,
    val availableFrom: String,
    val availableTo: String,
    val propertyId: Long,
    val propertyName: String,
    val city: String
)